src/coreclr/vm/nativelibrary.cpp

```diff
@@ -574,21 +576,16 @@ namespace
 
         int varCount = 0;
 
-        // The purpose of following code is to workaround LoadLibrary limitation:
-        // LoadLibrary won't append extension if filename itself contains '.'. Thus it will break the following scenario:
-        // [DllImport("A.B")] // The full name for file is "A.B.dll". This is common code pattern for cross-platform PInvoke
-        // The workaround for above scenario is to call LoadLibrary with "A.B" first, if it fails, then call LoadLibrary with "A.B.dll"
+        // The runtime will always append the '.dll' extension unless the name ends with a "."
+        // or with an existing known extension. This is done due to issues with case-sensitive file systems
+        // on Windows that only search for ".DLL" as opposed to the more common ".dll".
         auto it = libName.Begin();
-        if (!libNameIsRelativePath ||
```

Comment:
Why don't we care about relative versus absolute path anymore? It seems inconsistent with other platforms where we say if you give us an absolute path, we expect that is exactly what you want, so that is what we are using.

Reply:
I was trying to be as forgiving as possible, but your point is well-taken. I didn't see any of these guarantees in the `DllImport` doc so didn't really consider that point-of-view. The addition of extension seems to be a Windows loader feature that isn't part of `dlopen` so always adding it on Windows to handle the case-sensitive scenario and be consistent was prudent in my mind. Does that change your opinion or is keeping this consistent preferred?

Reply:
Doesn't the Windows loader only add the extension if the name is specified without the path? At least that is what the doc says - I did not actually check the behaviour.

Reply:
Related to this conversation, we should document the probing behavior we have on each platform for the next time we have to look at this: https://github.com/dotnet/runtime/issues/4596

Reply:
> Doesn't the Windows loader only add the extension if the name is specified without the path? At least that is what the doc says - I did not actually check the behaviour.

You would think. That is one of those behaviors I've mentioned in the past. Using ProcMon, you can see that if one supplies `C:\Users\foo`, probing will be for `C:\Users\foo.DLL`.

Reply:
Ech. I would still prefer the consistency though.

I think of the variation / extension probing as mainly helpful for cross-platform - otherwise you could just put in the full name. If you are using an absolute path, I would think it is unlikely you are doing it cross-plat. To me, adding the probing on Windows for absolute paths adds inconsistency in our cross-plat behaviour while not really unblocking users.

Reply:
We also call out that we don't do anything to absolute paths in https://docs.microsoft.com/dotnet/standard/native-interop/cross-platform